packages/workspace/src/core/project-graph/build-nodes/npm-packages.ts

```diff
@@ -0,0 +1,24 @@
+import * as stripJsonComments from 'strip-json-comments';
+import { ProjectGraphContext, AddProjectNode } from '../project-graph-models';
+
+export function buildNpmPackageNodes(
+  ctx: ProjectGraphContext,
+  addNode: AddProjectNode,
+  fileRead: (s: string) => string
+) {
+  const packageJson = JSON.parse(stripJsonComments(fileRead('package.json')));
+  const deps = {
+    ...packageJson.dependencies,
+    ...packageJson.devDependencies
+  };
+  Object.keys(deps).forEach(d => {
+    addNode({
+      type: 'npm',
+      name: d,
+      data: {
```

Comment:
Can we give it a `root` of the `require.resolve(deps[d])` relative from the workspace root?

For example, `@angular/core` should be `node_modules/@angular/core`

This way, the logic in https://github.com/nrwl/nx/pull/2195 will automatically resolve the dependencies?

It will resolve `@angular/core` to `node_modules/@angular/core/core.d.ts` which is inside of the project root above.

Reply:
+1